docs/0002-database-migration-errors.md

```diff
@@ -0,0 +1,81 @@
+# Handling Errors During Database Migration
+
+* Status: accepted
+* Date: 2021-06-08
+
+## Context and Problem Statement
+
+Database migrations have a relatively likely chance of hitting two different
+kinds of errors: database corruption errors and errors in the migration login.
+Corruption errors should be handled deleteing the database and restarting (see
+issue #2628), however migration logic errors should be handled by returning an
+error code and releasing fixed code.  What should we do if we see an error
+during the migration process?
+
+## Decision Drivers
+
+* Deleting valid user data should be avoided at almost any cost
+* Keeping a corrupted database around is almost as bad.  It currently prevents
+  the component from working at all.
+* There's no failproof way to distinguish the two error cases.
```

Comment:
While this is true, I think there are *some* cases that can be distinguished - eg, `SQLITE_CORRUPT`.